internal/southboundhandler/handler.go

```diff
@@ -335,13 +340,28 @@ func getMachineOwnerName(intelmachine *infrastructurev1alpha1.IntelMachine) (str
 	return "", errors.New("machine not found")
 }
 
-func extractBootstrapScript(secret *corev1.Secret, kind, providerID string) (string, error) {
-	format, ok := secret.Data["format"]
-	if !ok {
-		return "", errors.New("missing format in bootstrap secret")
+func providerIDCommands(configDir, providerID string) []cloudinit.Cmd {
+	filename := configDir + "providerID.yaml"
+	return []cloudinit.Cmd{
+		{
+			Cmd:  "mkdir",
+			Args: []string{"-p", configDir},
+		},
+		{
+			Cmd:   "/bin/sh",
+			Args:  []string{"-c", fmt.Sprintf("cat > %s /dev/stdin", filename)},
+			Stdin: fmt.Sprintf(`kubelet-arg+: [\"--provider-id=%s\"]`, providerID),
+		},
+		{
+			Cmd:  "chmod",
+			Args: []string{"0640", filename},
+		},
 	}
+}
 
-	if string(format) != "cloud-config" {
+func extractBootstrapScript(secret *corev1.Secret, kind, providerID string) (string, error) {
+	format, ok := secret.Data["format"]
+	if ok && string(format) != "cloud-config" {
```

Comment:
is "cloud-config" standard for any provider?

Reply:
It's standard among RKE2 and Kubeadm at least.  K3S seems to support only cloud-config format and does not specify the "format" field in the secret.